pkg/visitors/table.go

```diff
@@ -26,57 +31,105 @@ func (v *Table) Run(f uefi.Firmware) error {
 
 // Visit applies the Table visitor to any Firmware type.
 func (v *Table) Visit(f uefi.Firmware) error {
+	var offset uint64
 	switch f := f.(type) {
 	case *uefi.FlashImage:
-		return v.printRow(f, "Image", "", "")
+		v.depth = v.indent + 1
```

Comment:
depth should not be = v.indent + 1. If we ever want to add more cli's that go one or two levels deeper, this would cause it to always only go one level deep, it should be depth++ in uefi.FlashImage (because we want to not count the flashimage), and not anywhere else. Then in 

RegisterCLI("layout-table", "print out offset and size information of top level firmware volumes in a pretty table", 0, func(args []string) (uefi.Visitor, error) {
		return &Table{Layout: true, TopLevel: true, depth: 1}, nil
	})

This way we can easily add more cli's that traverse different depths I think

Reply:
I first though of using `Depth` in the interface ... and found it confusing:
If you ask for a depth of 2, you'll end with a depth of 4 on an Intel Image (2 + Image + IFD before BIOS Region)

So I opted for the `TopLevel` flag to be in the interface and the `depth` to only be an implementation internal... (but being in the same package makes the Exported/private less obvious).

I'll try your approach and see how it goes ;)

Reply:
Thanks, that works, I was probably missing the point that IFD is not a container, only uefi.FlashImage is...